client.go

```diff
@@ -1375,3 +1397,545 @@ func (c *ToznySDKV3) UnbrokerShare(ctx context.Context, authorizerClientID strin
+	if secret.SecretType == "" {
+		return errors.New("type cannot be empty")
+	}
+	if !SliceContainsString(SECRET_TYPES, secret.SecretType) {
+		return errors.New("invalid type")
+	}
+	if secret.SecretName == "" {
+		return errors.New("name cannot be empty")
+	}
+	matched, err := regexp.MatchString(`^[a-zA-Z0-9-_]{1,50}$`, secret.SecretName)
+	if err != nil {
+		return err
+	}
+	if !matched {
+		return errors.New("Secret name must contain 1-50 alphanumeric characters, -, or _")
+	}
+	if secret.SecretValue == "" && secret.SecretType == "Credential" {
+		return errors.New("Value cannot be empty")
+	}
+	if secret.SecretType == "File" && strings.TrimSpace(secret.SecretName) == "" {
+		return errors.New("File name cannot be empty")
+	}
+	if secret.SecretType == "Client" {
+		err := VerifyRawClientCredentials(secret.SecretValue)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// SliceContainsString checks if str is present in an array of strings
+func SliceContainsString(list []string, str string) bool {
+	for _, v := range list {
+		if v == str {
+			return true
+		}
+	}
+	return false
+}
+
+type ClientConfigRoot struct {
+	APIBaseURL        string      `json:"api_url"`
+	APIKeyID          string      `json:"api_key_id"`
+	APISecret         string      `json:"api_secret"`
+	ClientID          string      `json:"client_id"`
+	ClientEmail       string      `json:"client_email"`
+	PublicKey         string      `json:"public_key"`
+	PrivateKey        string      `json:"private_key"`
+	PublicSigningKey  string      `json:"public_signing_key"`
+	PrivateSigningKey string      `json:"private_signing_key"`
+	Version           interface{} `json:"version"`
+}
+
+// VerifyRawClientCredentials ensures that client credentials are in the proper form and contain all required keys
+func VerifyRawClientCredentials(credential string) error {
+	var clientConfig ClientConfigRoot
+	err := json.Unmarshal([]byte(credential), &clientConfig)
+	if err != nil {
+		return err
+	}
+	values := reflect.ValueOf(clientConfig)
+	keys := values.Type()
+	for index := 0; index < values.NumField(); index++ {
+		if keys.Field(index).Name != "ClientEmail" && values.Field(index).Interface() == "" {
+			return fmt.Errorf("Value for %s must be non-empty", keys.Field(index).Tag.Get("json"))
+		}
+	}
+	// Check that the version is either an integer or a string containing a number
+	_, versionInt := clientConfig.Version.(int)
+	_, err = strconv.Atoi(fmt.Sprintf("%v", clientConfig.Version))
+	if !versionInt && err != nil {
+		return errors.New("Value for version must be a number")
+	}
+	// ClientID must be a UUID
+	_, err = uuid.Parse(clientConfig.ClientID)
+	if err != nil {
+		return errors.New("Value for client_id should be in UUID format")
+	}
+	// Base64 encoded keys must have correct lengths
+	if len(clientConfig.PrivateSigningKey) != e3dbClients.Base64EncodedPrivateSigningKeyLength {
+		return errors.New("Invalid key length: private_signing_key")
```

Comment:
Lets have these errors tell us what the length was and what it should be. Then I don't need to dig into the code as much if I on accident swap my public and private keys.